src/Collection/CollectionTrait.php

```diff
@@ -605,6 +605,26 @@ public function zipWith($items, $callable)
      * {@inheritDoc}
      *
      */
+    public function chunk($chunkSize)
+    {
+        return $this->unfold(function ($v, $k, $iterator) use ($chunkSize) {
+            $values = [$v];
+            for ($i = 1; $i < $chunkSize; $i++) {
+                $iterator->next();
+                if (!$iterator->valid()) {
+                    break;
+                }
+                $values[] = $iterator->current();
+            }
+
+            return empty($values) ? $values : [$values];
```

Comment:
When would values be empty, if it always contains at least one element from line 611?